File: JavaScript/프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.js


function solution(arr1, arr2) {
    let n = arr1.length;
    let m = arr2[0].length;
    let answer = [...new Array(n)].map((_,i) => new Array(m).fill(0));
    const calc = (a, b) => {
        let sum_val = 0;
        for (let i = 0; i < arr1[0].length; i++){
            sum_val += arr1[a][i] * arr2[i][b];
        }
        return sum_val;
    }
    for (let i = 0; i < n ; i++){
        for (let j = 0; j < m; j++){
            answer[i][j] = calc(i, j);
        }
    }
    return answer;
}

File: C++/프로그래머스/2/12949. 행렬의 곱셈/행렬의 곱셈.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr1, vector<vector<int>> arr2) {
    vector<vector<int>> answer;
    int arr1_row = arr1.size();
    int arr1_col = arr1[0].size();
    int arr2_row = arr2.size();
    int arr2_col = arr2[0].size();
    // answer = vector<vector<int>>(arr1_row, vector<int>(arr2_col, 0));
    answer.resize(arr1_row, vector<int>(arr2_col, 0));
    for (int i = 0; i < arr1_row; i++){
        for (int j = 0; j < arr2_col; j++){
            int sum_val = 0;
            for (int k = 0; k < arr1_col; k++){
                sum_val += arr1[i][k] * arr2[k][j];
            }
            answer[i][j] = sum_val;
        }
    }
    
    return answer;
}